boa/src/builtins/error/eval.rs

```diff
@@ -57,17 +58,27 @@ impl EvalError {
 
     /// Create a new error object.
     pub(crate) fn constructor(
-        this: &Value,
+        new_target: &Value,
         args: &[Value],
         context: &mut Context,
     ) -> Result<Value> {
+        let prototype = match new_target {
+            Value::Object(obj) => match obj.get(&PROTOTYPE.into(), context)? {
+                Value::Object(ref o) => o.clone(),
+                _ => context.standard_objects().error_object().prototype(),
+            },
```

Comment:
Not sure if this would be more clear:
```suggestion
            Value::Object(obj) => obj.get(&PROTOTYPE.into(), context)?.as_object().unnwrap_or_else(|| context.standard_objects().error_object().prototype()),
```

It would be applicable to the rest of the places where this happens, and we could also use it for the `new_target` match:

```rust
let prototype = new_target.as_object().map(|obj| {
    obj.get(&PROTOTYPE.into(), context)?.as_object()
}).flatten().unwrap_or_else(|| context.standard_objects().error_object().prototype());
```

This would add an extra pointer clone (which would probably be negligible and maybe optimized out). Didn't check the code, but using the [combinators](https://doc.rust-lang.org/stable/std/option/enum.Option.html) of the `Option` type many times gives a more idiomatic/clean code.